src/Registration/cReg/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+#========================================================================
+# Author: Richard Brown
+# Copyright 2016, 2017 University College London
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+
+# Add in the common files
+SET(SOURCES "SIRFReg.cpp" "SIRFRegNiftyAladinSym.cpp" "SIRFRegNiftyF3dSym.cpp" "SIRFRegImageWeightedMean.cpp" "SIRFRegNiftyResample.cpp" "SIRFRegMisc.cpp" "SIRFRegTransformation.cpp" "SIRFRegAffineTransformation.cpp")
+SET(HEADERS  "SIRFReg.h"   "SIRFRegNiftyAladinSym.h"   "SIRFRegNiftyF3dSym.h"   "SIRFRegImageWeightedMean.h"   "SIRFRegNiftyResample.h"   "SIRFRegMisc.h"    "SIRFRegTransformation.h"   "SIRFRegAffineTransformation.h" "SIRFRegParser.h"   "SIRFRegParserKey.h" )
+
+# Add in nifti_image wrappers
+SET(SOURCES ${SOURCES} "NiftiImageData.cpp" "NiftiImageData3D.cpp" "NiftiImageData3DTensor.cpp" "NiftiImageData3DDeformation.cpp" "NiftiImageData3DDisplacement.cpp")
+SET(HEADERS ${HEADERS}  "NiftiImageData.h"   "NiftiImageData3D.h"   "NiftiImageData3DTensor.h"   "NiftiImageData3DDeformation.h"   "NiftiImageData3DDisplacement.h")
+
+# If we're also wrapping to python or matlab, include the c-files
+IF(BUILD_PYTHON OR BUILD_MATLAB)
+	SET(SOURCES ${SOURCES} "csirfreg.cpp" "csirfreg_p.cpp")
+	SET(HEADERS ${HEADERS}  "csirfreg.h"   "csirfreg_p.h" )
+ENDIF()
+
+# Create library
+ADD_LIBRARY(SIRFReg ${SOURCES})
+#TARGET_LINK_LIBRARIES(SIRFReg SIRF_common)
+
+# Link against stir and gadgetron
+TARGET_LINK_LIBRARIES(SIRFReg cstir)
+#TARGET_LINK_LIBRARIES(SIRFReg cgadgetron)
+
+# Boost
+FIND_PACKAGE(Boost 1.63.0 COMPONENTS filesystem)
```

Comment:
I think this needs system as well.
https://stackoverflow.com/questions/9723793/undefined-reference-to-boostsystemsystem-category-when-compiling